toolchain/parse/tree.h

```diff
@@ -277,11 +279,18 @@ class Tree : public Printable<Tree> {
   auto Verify() const -> ErrorOr<Success>;
 
   // Like ExtractAs(), but malformed tree errors are not fatal. Should only be
-  // used by `Verify()`.
+  // used by `Verify()` or by tests.
   template <typename T>
   auto VerifyExtractAs(NodeId node_id, ErrorBuilder* trace) const
       -> std::optional<T>;
 
+  // Sets the kind of a node. This is intended to allow putting the tree into a
+  // state where verification can fail, in order to make the failure path of
+  // `Verify` testable.
+  auto SetNodeKindForVerifyTest(NodeId node_id, NodeKind kind) -> void {
+    node_impls_[node_id.index].kind = kind;
+  }
```

Comment:
There's the intention to have things like the parse tree be an API for tools. Would it be worth using `friend` to keep this private so that it's not part of the public API?

Reply:
I don't have any particular opinion here. https://abseil.io/tips/135 suggests adding a test-only public member function instead of a friend, but I'm happy to go down the friend path if you prefer. I don't know how much that tip considered the Hyrum's law impact of making test-only functionality like this public.

Reply:
FWIW I lean down the Hyrum's law path here if we consider this a public API -- I would expect even abseil doesn't exposing its internal-test-only things as public. :)

I think the test peer approach noted in the tip might be fine. I would note the "and other clients (not just the test) might find them useful" caveat -- the intent should be that this is a public API because it's a public API, not just for our own internal tests.

Reply:
Done.